poh/src/poh_recorder.rs

```diff
@@ -342,91 +406,53 @@ impl PohRecorder {
         }
     }
 
-    fn reached_leader_tick(&self, leader_first_tick_height_including_grace_ticks: u64) -> bool {
-        let target_tick_height = leader_first_tick_height_including_grace_ticks.saturating_sub(1);
-        let ideal_target_tick_height = target_tick_height.saturating_sub(self.grace_ticks);
-        let next_tick_height = self.tick_height.saturating_add(1);
-        let next_slot = self.slot_for_tick_height(next_tick_height);
+    fn reached_leader_tick(&self) -> bool {
+        let leader_tick_heights = match self.leader_tick_heights.as_ref() {
+            Some(leader_tick_heights) => leader_tick_heights,
+            None => return false,
+        };
+
         // We've approached target_tick_height OR poh was reset to run immediately
         // Or, previous leader didn't transmit in any of its leader slots, so ignore grace ticks
-        self.tick_height >= target_tick_height
-            || self.start_tick_height + self.grace_ticks
-                == leader_first_tick_height_including_grace_ticks
-            || (self.tick_height >= ideal_target_tick_height
-                && (self.prev_slot_was_mine(next_slot)
-                    || !self.is_same_fork_as_previous_leader(next_slot)))
+        leader_tick_heights.target_tick_reached()
```

Comment:
Wondering if this is actually functionally equivalent to the original code `self.tick_height >= target_tick_height`, or if there's an off by one error.

This is because in the original code `target_tick_height` = `leader_first_tick_height_including_grace_ticks.saturating_sub(1);` which is equal to `leader_first_tick_height + grace_ticks - 1` (note I don't recall why the -1 here is necessary).

However in the new code `target_tick_reached()` is true if we're `>= leader_first_tick_height + grace_ticks`.

Reply:
It's the same, `target_tick_reached` is true when `target_tick_height_reached_at` has been recorded and that's recorded when `tick_height + 1 >= leader_first_tick_height + grace_ticks`. That's equivalent to `tick_height >= leader_first_tick_height + grace_ticks - 1`

Reply:
The -1 was necessary before because we were really checking `tick_height + 1 >= leader_first_tick_height + grace_ticks` and this refactor makes that more clear

Reply:
that makes sense, missed the `+1` in setting `target_tick_reached`